packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -603,29 +665,84 @@ export class LinkExtension extends MarkExtension<LinkOptions> {
             }
 
             const nodeText = tr.doc.textBetween(pos, pos + node.nodeSize, undefined, ' ');
-            this.findAutoLinks(nodeText)
+
+            const foundAutoLinks: FoundAutoLink[] = [];
+
+            for (const match of findMatches(nodeText, new RegExp(/(\S)+/, 'gm'))) {
+              const matchedText = getMatchString(match);
+
+              // Skip if URL doesn't have minimum length
+              // TODO: Could make this configurable
+              if (matchedText.length < MIN_LINK_LENGTH) {
+                continue;
+              }
+
+              const matchStart = match.index;
+              const matchEnd = matchStart + matchedText.length;
+              const urlInMatchRange = this.findURL(matchedText);
+              const linkMarkInRange = this.getLinkMarksInRange(doc, matchStart, matchEnd, true)[0];
+
+              // Don't update existing links that are not in selection range
+              if (
+                linkMarkInRange &&
+                urlInMatchRange &&
+                (linkMarkInRange.to >= to ||
+                  !within(from, linkMarkInRange.from, linkMarkInRange.to)) &&
+                linkMarkInRange.mark.attrs.href === this.buildHref(urlInMatchRange)
+              ) {
+                continue;
+              }
+
+              // Skip if no valid link is in the match
+              if (!urlInMatchRange && !new RegExp('^[0-9|-]+$').test(matchedText)) {
```

Comment:
Can you explain the intent of this regex here, I'm not sure what were asserting?

I suspect the vertical bar in the character group here is a mistake? See [here](https://regexper.com/#%5E%5B0-9%7C-%5D%2B%24)

P.s. we should also store this in a const

Reply:
This was from a previous iteration, checking for phone numbers. Thanks for calling it out - consider it removed